nox/manifest.py

```diff
@@ -172,9 +172,23 @@ def filter_by_keywords(self, keywords: str) -> None:
                 session names are checked against.
         """
         self._queue = [
-            x for x in self._queue if keyword_match(keywords, x.signatures + [x.name])
+            x
+            for x in self._queue
+            if keyword_match(keywords, x.signatures + x.tags + [x.name])
         ]
 
+    def filter_by_tags(self, tags: list[str]) -> None:
+        """Filter sessions using pytest-like tag expressions.
```

Comment:
I wonder if this was refactored at a point. This comment makes this all seem a bit more powerful than it seems to be.

1) What is pytest like about this? As best I can tell tags don't support additional keywords and just checks whether or not any of the specified tags are present?

2) When I read 'Python expression' I start to think some sort of eval() is going on, but we are just comparing strings right?

Reply:
That's correct. This filter just matches by the raw tags. It's the keywords filter that supports tag expressions.

Reply:
In that case I think we should change the docstring to reflect that e.g. `"Filter sessions on matching tags"`

Reply:
@FollowTheProcess yup, I'd changed it to `"Filter sessions by their tags"`. Let me know if that's more like it.

Reply:
That's great, thanks